src/plugins/intel_cpu/src/utils/debug_capabilities.h

```diff
@@ -15,12 +15,71 @@
 
 #include <onednn/dnnl.h>
 #include <dnnl_debug.h>
+#include <cpu/x64/jit_generator.hpp>
 
 #include "openvino/core/model.hpp"
 
 namespace ov {
 namespace intel_cpu {
 
+class RegPrintManager;
+
+#define PRINT_REGISTER_SELECTOR(_1, _2, name, ...) name
+
+#define PRINT_REGISTER_WITH_VALUE(REG_VAL)             \
+{                                                      \
+    RegPrintManager manager(*this, REG_VAL, #REG_VAL); \
+    manager.PrintRegister(); \
+}
+
+#define PRINT_REGISTER_WITH_VALUE_PRECISION(REG_VAL, REG_PRECISION)   \
+{                                                                     \
+    RegPrintManager manager(*this, REG_VAL, #REG_VAL, REG_PRECISION); \
+    manager.PrintRegister();                                          \
+}
+
+#define PRINT_REGISTER(...)                                      \
+    PRINT_REGISTER_SELECTOR(__VA_ARGS__,                         \
+                            PRINT_REGISTER_WITH_VALUE_PRECISION, \
+                            PRINT_REGISTER_WITH_VALUE)(__VA_ARGS__)
+
+class RegPrintManager {
+public:
+    RegPrintManager(dnnl::impl::cpu::x64::jit_generator &h, const Xbyak::Reg &reg_val, const char *reg_name): h(h),
+                    reg_val(reg_val), reg_name(reg_name), reg_dt(dnnl::memory::data_type::undef) {
+        InitDefaultPrecision();
+    }
+    RegPrintManager(dnnl::impl::cpu::x64::jit_generator &h, const Xbyak::Reg &reg_val, const char *reg_name,
+                    dnnl::memory::data_type reg_dt) : h(h), reg_val(reg_val), reg_name(reg_name), reg_dt(reg_dt) {
+        InitDefaultPrecision();
+    }
+    RegPrintManager(dnnl::impl::cpu::x64::jit_generator &h, const Xbyak::Xmm &xmm_val, const char *reg_name): h(h),
```

Comment:
I would reccomed to remove ability to print regiters with default precision, since it creates more confusion. Lets force the user to explicilty specify desired precision by template argument as I mentioned above.

Reply:
Fixed. Thanks Dmitry!